terraform/slurm_cluster/modules/slurm_nodeset_tpu/main.tf

```diff
@@ -90,17 +87,10 @@ data "google_compute_default_service_account" "this" {
 }
 
 data "google_compute_subnetwork" "nodeset_subnetwork" {
-  count = local.snetwork_valid ? 1 : 0
+  name = var.subnetwork
 
-  project = var.project_id
-  region  = local.region
-  name    = var.subnetwork
-  self_link = (
-    length(regexall("/projects/([^/]*)", var.subnetwork)) > 0
-    && length(regexall("/regions/([^/]*)", var.subnetwork)) > 0
-    ? var.subnetwork
-    : null
-  )
+  project   = var.project_id
+  self_link = var.subnetwork
```

Comment:
`var.subnetwork` will be always valid self_link in context of GHPC invocation, but it may break non-GHPC usage.
Let's remove it until we change semantic `var.subnetwork -> var.subnetwork_self_link`

Reply:
Offline discussion: There might be some issues with removing self_link here. So decided to keep it for now.